### DISCUSSION TITLE
What is Acc. Battery Level?

### DISCUSSION THREAD

**Discussion by ruckusvol** [BimmerGestalt/AAIdrive#570] [Q&A]
What is the data in the field for Acc. Battery Level? Mine says 57%.

**Comment by hufman** [comment]
The Acc. Battery Level is loaded from the Car Data Service with [some slight formatting applied](https://github.com/BimmerGestalt/AAIdrive/blob/main/app/src/main/java/me/hufman/androidautoidrive/phoneui/viewmodels/CarDrivingStatsModel.kt#L200-L203). The [CDS data point](https://bimmergestalt.github.io/BMWConnectedAnalysis/cds/) for `sensors.battery` only has a single number, which I guess to be a percentage. I don't know how the car calculates this number, however.

**Reply by ruckusvol** [reply]
It looks like it's either the State of Charge (SoC) or the State of Health (SoH).  Both are expressed in percentage.  From some info I found:

The battery SoC defines the level of charge possessed by a battery at any given point of time and is indicative of the available energy that can be retrieved. Similarly, the SoH defines the ability of the battery to store charge at a given SoC relative to a new battery. BMS can collect external data on the cell’s voltage, current and temperatures and use its built-in computational algorithm to diagnose and estimate SoC and SoH.

It would be really nice to figure out whether than number is the SoH or the SoC. My assumption is that the SoC would change much more rapidly than SoH and could probably be tested by sitting in the driveway with ignition on but the car not running and having the AC, Stereo, lights, etc on for a bit.

Do you know if there's any way to figure out from the coding side which it is?  Why is it labeled Acc Battery Level in AAIDrive (what's the Acc for?)?  I'm happy to help figure it out if it's helpful to you guys.

**Reply by hufman** [reply]
I'd gladly welcome any further testing! I guessed it to be the Accessory Battery, as opposed to the Traction Battery. My car is an EV, and so I would rely on you to test whether it decreases during accessory mode. I'm unfamiliar with coding, and so I'm not sure how to correlate any information found there with this CDS value, but would be interested to hear what you find!

**Reply by ruckusvol** [reply]
Ok, I did some testing.  I’m almost certain it’s the State of Charge (SoC).  I turned on the air, stereo, seat warmers, etc and watched the number drop steadily by a point or two per minute.  It has to be the charge level.  For gas vehicles I would call it Battery Charge or Battery Level.  But it’s not tied to accessories since there’s only 1 battery in the vehicle.

I don’t know how it applies to you with an electric vehicle.  I assume you want to name it something that works for both types.  If EVs do call it SoC then you could title it Battery SoC but I’m assuming that isn’t the case.  What do you think?

**Reply by hufman** [reply]
I name the traction battery's SoC as "Electric Charge", and the car also provides an Electric Range stat that the app shows.
Since accessories are solely powered by the 12V battery, the current name of Acc Battery Level seems correct, but what name are you thinking it should be?

**Reply by ruckusvol** [reply]
I pulled up the manual for the i3 and it refers to the two batteries as the high-voltage battery and the vehicle battery.  The vehicle battery is the 12V.  On my regular gas-powered car it’s called the vehicle battery as well.  Calling it the vehicle battery would be technically correct but may be confusing for people with EVs like you.

What about for the vehicle battery SoC calling it “12V Battery Charge”?  It’s expressed in percentage and then should be accurate for both vehicle types.  Maybe there’s another good option, but I think Acc Battery Charge is confusing.  It sent me looking in the manuals if there was some small auxiliary battery or something.
